xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,423 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "VideoStreamSelectHelper.h"
+
+#include "FileItem.h"
+#include "FileItemList.h"
+#include "ServiceBroker.h"
+#include "application/ApplicationComponents.h"
+#include "application/ApplicationPlayer.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "utils/LangCodeExpander.h"
+#include "utils/StreamDetails.h"
+#include "utils/StringUtils.h"
+#include "utils/log.h"
+
+namespace
+{
+constexpr int FILEITEM_INDEX_DISABLE = -2; // FileItem index value to disable the stream
+constexpr int FILEITEM_INDEX_NONE = -1; // FileItem index value for "none" stream
+
+// \brief Make a FileItem entry with "Disable" label, allow to disable a stream
+CFileItemPtr MakeFileItemDisable(const bool isSelected)
```

Comment:
I suggest not to use the `typedef`, but the real type which is `std::shared_ptr<CFileItem>`. This is what we do with new code. Please change this here, and everywhere else in this file.

`const`makes no sense for primitive types when used as stack parameters. Remove the `const`, please.

Reply:
i usually add const to arguments like this isSelected because make in clear that the value will be not modified by the method code
if you remove the const the value can be modified by method

Reply:
which for by value parameters is pointless. The caller simply does not need to know that the implementation does not modify the stack param.

Reply:
https://stackoverflow.com/questions/8714250/isnt-const-redundant-when-passing-by-value

Reply:
it's not about caller, it's more about readability,
in this case the code is small and can also be removed,
but if you have a method with very long code, clearly state that the variable will not be modified by the method code, as described on the linked url

Reply:
One more time: the const is not part of the contract. It means nothing to the *caller* whether it is present or not. You describe the contract with the method declaration , not implementation details. 

BUT, making the param const enforces the *implementation* to make a copy of the stack parameter which is already a copy in case the implementation wants to manipulate the parameters value. Thus you are taking the importation the chance to optimize the code for no reason.

Reply:
now i understood, i will do it